src/commonMain/kotlin/exchange.dydx.abacus/state/app/helper/TriggerOrderToastGenerator.kt

```diff
@@ -0,0 +1,380 @@
+package exchange.dydx.abacus.state.app.helper
+
+import exchange.dydx.abacus.output.PerpetualState
+import exchange.dydx.abacus.output.PositionSide
+import exchange.dydx.abacus.output.SubaccountOrder
+import exchange.dydx.abacus.output.input.OrderStatus
+import exchange.dydx.abacus.output.input.OrderType
+import exchange.dydx.abacus.output.isOppositeOf
+import exchange.dydx.abacus.protocols.FormatterProtocol
+import exchange.dydx.abacus.protocols.LocalizerProtocol
+import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.protocols.PresentationProtocol
+import exchange.dydx.abacus.protocols.ThreadingProtocol
+import exchange.dydx.abacus.protocols.ThreadingType
+import exchange.dydx.abacus.protocols.Toast
+import exchange.dydx.abacus.protocols.ToastType
+import exchange.dydx.abacus.responses.ParsingError
+import exchange.dydx.abacus.state.manager.HumanReadableCancelOrderPayload
+import exchange.dydx.abacus.state.manager.HumanReadablePlaceOrderPayload
+import exchange.dydx.abacus.state.manager.HumanReadableTriggerOrdersPayload
+import exchange.dydx.abacus.utils.IList
+import exchange.dydx.abacus.utils.Logger
+
+interface TriggerOrderToastGeneratorProtocol {
+    fun onTriggerOrderSubmitted(subaccountNumber: Int, payload: HumanReadableTriggerOrdersPayload, state: PerpetualState?)
+    fun onTriggerOrderResponse(subaccountNumber: Int, successful: Boolean, error: ParsingError?, data: Any?)
+}
+
+class TriggerOrderToastGenerator(
+    private val presentation: PresentationProtocol?,
+    private val parser: ParserProtocol,
+    private val formatter: FormatterProtocol?,
+    private val localizer: LocalizerProtocol?,
+    private val threading: ThreadingProtocol?,
+) : TriggerOrderToastGeneratorProtocol {
+    private enum class TriggerOrderStatus {
+        Submitting,
+        Success,
+        Failed
+    }
+
+    private var orderPayloadStatus: MutableMap<String, TriggerOrderStatus> = mutableMapOf()
+
+    private var state: PerpetualState? = null
+
+    override fun onTriggerOrderSubmitted(
+        subaccountNumber: Int,
+        payload: HumanReadableTriggerOrdersPayload,
+        state: PerpetualState?,
+    ) {
+        if (presentation == null) {
```

Comment:
Perhaps instead of taking in a nullable `PresentationProtocol`, we could just make a no-op implementation of the generator?

Reply:
We could.. there is no clear advantage of one way or another.